### PR TITLE
devops(flakiness): azure function to store dashboard in a new format

### PR DESCRIPTION
**Pull request by aslushnikov**
This refactors azure function and adds a new dashboard format implementation that is 15x smaller while tracking specs with non-empty annotations.

Old dashboard is still updated.

### CODE REVIEW

File: utils/flakiness-dashboard/processing/utils.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      blobHTTPHeaders: {
+        blobContentEncoding: 'gzip',
+        blobContentType: 'application/json; charset=UTF-8',
+      }
+    });
+  }
+}
+
+async function deleteBlob(container, blobName) {
+  // First we do - delete the blob.
```
Comment (JoelEinbinder) [inline utils/flakiness-dashboard/processing/utils.js:75]:
```suggestion
```
I think first we delete the blob, last we delete the blob. In a function called delete blob. I think this comment is not needed :).

File: utils/flakiness-dashboard/processing/index.js [hunk 2]
```diff
@@ -14,160 +14,22 @@
...
-      URL: ${report.metadata.runURL}
-      timestamp: ${report.metadata.commitTimestamp}
-      added specs: ${addedSpecs}
-    ===== complete in ${Date.now() - timestamp}ms =====
-    `);
-  } catch (e) {
-    context.log(e);
-    return;
-  }
+  // Uplaod report to both dashboards.
```
Comment (JoelEinbinder) [inline utils/flakiness-dashboard/processing/index.js:29]:
```suggestion
  // Upload report to both dashboards.
```

**Comment by JoelEinbinder** [comment]
When do we stop updating the v1 dashboard?

File: utils/flakiness-dashboard/processing/dashboard_v2.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {SimpleBlob, flattenSpecs} = require('./utils.js');
+
+const DASHBOARD_VERSION = 1;
```
Comment (JoelEinbinder) [inline utils/flakiness-dashboard/processing/dashboard_v2.js:19]: should this not be 2?
```suggestion
const DASHBOARD_VERSION = 2;
```

**Review by JoelEinbinder — APPROVED** [review]

File: utils/flakiness-dashboard/processing/dashboard_v2.js (hunk 3 above)
Comment (aslushnikov) [reply utils/flakiness-dashboard/processing/dashboard_v2.js:19]: v2 dashboard is side-by-side with the v1 dashboard in a separate blob - so no changes are necessary

**Comment by aslushnikov** [comment]
> When do we stop updating the v1 dashboard?

Once front-end is migrated to the new format and it proves to be working just fine :-)

**Pull request merged**